game-core/src/main/java/games/strategy/engine/data/changefactory/AddUnits.java

```diff
@@ -36,11 +53,36 @@ public Change invert() {
   @Override
   protected void perform(final GameData data) {
     final UnitHolder holder = data.getUnitHolder(name, type);
-    holder.getUnitCollection().addAll(units);
+    final Collection<Unit> unitsWithCorrectOwner = buildUnitsWithOwner(data);
+    holder.getUnitCollection().addAll(unitsWithCorrectOwner);
```

Comment:
A test of some sort to verify the right functionality would be a good thing to have. If we change & break this, we'll have the extra safety net. Testing is really good for ensuring correctness, which is a further safety net that would be good to have as well.